src/DIRAC/WorkloadManagementSystem/JobWrapper/JobWrapperOfflineTemplate.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+""" This template will become the job wrapper that's actually executed.
+
+The JobWrapperOfflineTemplate is completed and invoked by the PushJobAgent and uses functionalities from JobWrapper module.
+It is executed in environment where external connections are not allowed.
+"""
+import hashlib
+import sys
+import json
+import os
+
+sitePython = os.path.realpath("@SITEPYTHON@")
+if sitePython:
+    sys.path.insert(0, sitePython)
+
+from DIRAC.Core.Base.Script import Script
+
+Script.parseCommandLine()
+
+from DIRAC import gLogger
+from DIRAC.WorkloadManagementSystem.JobWrapper.JobWrapper import JobWrapper
+
+os.umask(0o22)
+
+
+def execute(jobID: str, arguments: dict):
+    """The only real function executed here"""
+    payloadParams = arguments.pop("Payload", {})
+    if not payloadParams:
+        return 1
+
+    if not "PayloadResults" in arguments["Job"] or not "Checksum" in arguments["Job"]:
+        return 1
+
+    try:
+        job = JobWrapper(jobID)
+        job.initialize(arguments)  # initialize doesn't return S_OK/S_ERROR
+    except Exception as exc:  # pylint: disable=broad-except
+        gLogger.exception("JobWrapper failed the initialization phase", lException=exc)
+        return 1
+
+    payloadResult = job.process(**payloadParams)
+    if not payloadResult["OK"]:
+        return 1
+
+    # Store the payload result
+    with open(arguments["Job"]["PayloadResults"], "w") as f:
+        json.dump(payloadResult, f)
+
+    # Generate the checksum of the files present in the current directory
+    checksums = {}
+    for file in os.listdir("."):
+        if os.path.isfile(file):
+            hash_md5 = hashlib.md5()
```

Comment:
In python 3.11 there is https://docs.python.org/3/library/hashlib.html#file-hashing